foojay-resolver/src/main/kotlin/org/gradle/toolchains/foojay/distributions.kt

```diff
@@ -110,6 +94,44 @@ fun parseDistributions(json: String): List<Distribution> {
     return Gson().fromJson(json, DistributionsResult::class.java).result
 }
 
+
+private fun createVendorAliases(): Map<JvmVendorSpec, String> {
+    val tmpMap = mutableMapOf(
+        JvmVendorSpec.ADOPTIUM to "Temurin",
+        JvmVendorSpec.ADOPTOPENJDK to "AOJ",
+        JvmVendorSpec.AMAZON to "Corretto",
+        JvmVendorSpec.AZUL to "Zulu",
+        JvmVendorSpec.BELLSOFT to "Liberica",
+        JvmVendorSpec.IBM to "Semeru",
+        JvmVendorSpec.ORACLE to "Oracle OpenJDK",
+        JvmVendorSpec.SAP to "SAP Machine"
+    )
+    addPossiblyUndefinedVendor("IBM_SEMERU", "Semeru", tmpMap)
+    return tmpMap.toMap()
+}
+
+private fun createJ9AliasesMap(): Map<JvmVendorSpec, String> {
+    val tmpMap = mutableMapOf(
+        JvmVendorSpec.IBM to "Semeru",
+        JvmVendorSpec.ADOPTOPENJDK to "AOJ OpenJ9"
+    )
+    addPossiblyUndefinedVendor("IBM_SEMERU", "Semeru", tmpMap)
+    return tmpMap.toMap()
+}
+
+private fun addPossiblyUndefinedVendor(
+    vendorFieldName: String,
+    vendorAlias: String,
+    map: MutableMap<JvmVendorSpec, String>
+) {
+    try {
+        val vendorField = JvmVendorSpec::class.java.getDeclaredField(vendorFieldName)
+        map.put(vendorField.get(null) as JvmVendorSpec, vendorAlias)
+    } catch (_: Exception) {
+        // Ignore - removed in the current Gradle version
```

Comment:
What is "current". I wound suggest to revert this back to gradle 9. Current is dynamic and is even different from the point of the reader. Gradle if this plugin? Gradle of the consumer of this plugin..?

Reply:
I updated the sentence, but this is designed to be used for potentially other removals in the future. So it should not reference a particular version.